File: code/test_suites/cpp/FunctionBuildTest/src/function_pandas_fill_missing.cpp
#include <string>
#include <cmath>
#include <iomanip>
#include <sstream>

class FillMissingValues {
public:
    std::string fill_missing_values(double value_A, double value_B) {
        if (std::isnan(value_A) && std::isnan(value_B)) {
            return "A: nan, B: nan";
        } else {
            double meanA = std::isnan(value_A) ? NAN : value_A;
            double meanB = std::isnan(value_B) ? NAN : value_B;

            double finalA = std::isnan(meanA) ? NAN : meanA;
            double finalB = std::isnan(meanB) ? NAN : meanB;

            std::string resultA = std::isnan(finalA) ? "nan" : std::to_string(finalA);
            std::string resultB = std::isnan(finalB) ? "nan" : std::to_string(finalB);

            std::ostringstream oss;
            oss << "A: " << resultA << ", B: " << resultB;
            return oss.str();
        }
    }
};

File: code/generate_strategies/RA(name)/java/java/function_pandas_fill_missing.java
package com.example;

import tech.tablesaw.api.DoubleColumn;
import tech.tablesaw.api.Table;

public class FillMissingValues {
    public String fill_missing_values(Double value_A, Double value_B) {
        Table table = Table.create("data");

        // 创建列并添加数据
        DoubleColumn colA = DoubleColumn.create("A", 1);
        DoubleColumn colB = DoubleColumn.create("B", 1);
        colA.set(0, value_A);
        colB.set(0, value_B);
        table.addColumns(colA, colB);

        if (value_A == null && value_B == null) {
            return "A: nan, B: nan";
        } else {
            // 计算非null值的平均值
            double meanA = value_A != null ? value_A : Double.NaN;
            double meanB = value_B != null ? value_B : Double.NaN;

            // 使用平均值填充null值
            double finalA = Double.isNaN(meanA) ? Double.NaN : meanA;
            double finalB = Double.isNaN(meanB) ? Double.NaN : meanB;

            String resultA = Double.isNaN(finalA) ? "nan" : String.valueOf(finalA);
            String resultB = Double.isNaN(finalB) ? "nan" : String.valueOf(finalB);

            return String.format("A: %s, B: %s", resultA, resultB);
        }
    }
}